app/actions/views/channel.js

```diff
@@ -73,6 +75,26 @@ export function loadChannelsByTeamName(teamName) {
     };
 }
 
+export function loadPublicAndArchivedChannels(teamId, page, perPage, shouldLoadArchivedChannels) {
+    return async (dispatch) => {
+        return dispatch(getChannels(
+            teamId,
+            page,
+            perPage
+        )).then(async (publicChannels) => {
+            if (shouldLoadArchivedChannels) {
+                const archivedChannels = await dispatch(getArchivedChannels(
+                    teamId,
+                    page,
+                    perPage
+                ));
+                return archivedChannels;
```

Comment:
if `shouldLoadArchivedChannels` is true it will only return archived channels, is that what you're aiming for ?

Reply:
to me this should be something like:
```
1. get channels
2. get archived channels only when applies
3. return both public and archived channels in a `{data}` object or an error when needed
```